source/containers/DynamicCardRenderer/DynamicCardRenderer.tsx

```diff
@@ -73,6 +90,12 @@ const renderCardComponent = (component: CardComponent, navigation: any) => {
   if (component.type === 'button') {
     const { icon, iconPosition, text } = component;
     let onClick: () => void = () => null;
+
+    // treat info-modal separately since it doesn't fit the same pattern as the other buttons. 
+    if (component.action === 'infoModal') {
+      return <InfoButton {...component} />
+    }
+
     switch (component.action) {
```

Comment:
Out of the scope of this PR: But this switch case feels very redundant, since we define four functions for one onClick function. Why not define a onClick handler?
```
function handleClick() {
  switch (component.action) {
      case 'email':
        launchEmail(component.email)
        break;
      case 'phone':
        launchPhone(component.phonenumber)
        break;
      case 'navigate':
        if (navigation?.navigate) navigation.navigate(component.screen) // TODO think about sending parameters here
        break;
      case 'url':
        Linking.openURL(component.url);
        break;
    }
}
```

Reply:
Looking at it, I agree, that would be nicer.

Reply:
Refactored, since you pretty much already wrote the code for it.